kernel/include/kswap.h

```diff
@@ -117,6 +117,7 @@ static ALWAYS_INLINE unsigned int do_swap(unsigned int key,
 		wait_for_switch(new_thread);
 		arch_switch(new_thread->switch_handle,
 			     &old_thread->switch_handle);
+		sys_trace_thread_switched_in();
```

Comment:
commit message mentions x86_64, but this is in common code and applies to all arches using `CONFIG_SWITCH`